On 6/3/16 6:51 AM, Marcel Schneider wrote:
> |Traceback (most recent call last): File "RunPromptRecoCfg.py", line
> 111226, in <module> process.patMetUncertaintySequenceNoHF =
> cms.Sequence(process.ak4PFCHSL1FastL2L3CorrectorChain+process.ak4PFCHSL1FastL2L3ResidualCorrectorChain++process.patPFMetT1NoHFpatMetUncertaintySequenceNoHF+process.patPFMetT1SmearNoHFpatMetUncertaintySequenceNoHF)
> TypeError: bad operand type for unary +: 'Sequence' |
> 
> I have honestly no idea what this sequence does, but to me this is a
> sign that it is not DQM related.

Probably not, but I guess the only way is to check in a  test build with
the phase1 DQM changes reverted


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/14760#issuecomment-223584126>, or
> mute the thread
> <https://github.com/notifications/unsubscribe/AEdcbt8LXiCogN4Bykb8VKx8qmFJEKjWks5qIDFvgaJpZM4ItYgj>.
>

--- a/DQM/SiPixelPhase1Common/src/GeometryInterface.cc
+++ b/DQM/SiPixelPhase1Common/src/GeometryInterface.cc
@@ -212,7 +212,7 @@ void GeometryInterface::loadFromTopology(edm::EventSetup const& iSetup, const ed
   addExtractor(intern("PXForward"), pxforward, 0, 0);
 
   // Redefine the disk numbering to use the sign
-  auto& pxendcap = extractors[intern("PXEndcap")];
+  auto pxendcap = extractors[intern("PXEndcap")];
   auto diskid = intern("PXDisk");
   auto pxdisk = extractors[diskid];
   extractors[diskid] = [pxdisk, pxendcap] (InterestingQuantities const& iq) {
@@ -228,8 +228,8 @@ void GeometryInterface::loadFromTopology(edm::EventSetup const& iSetup, const ed
   assert(trackerGeometryHandle.isValid());
   
   // We need to track some extra stuff here for the Shells later.
-  auto& pxlayer  = extractors[intern("PXLayer")];
-  auto& pxladder = extractors[intern("PXLadder")];
+  auto pxlayer  = extractors[intern("PXLayer")];
+  auto pxladder = extractors[intern("PXLadder")];
   std::vector<Value> maxladders;
 
   // Now travrse the detector and collect whatever we need.
@@ -260,7 +260,7 @@ void GeometryInterface::loadFromTopology(edm::EventSetup const& iSetup, const ed
   // of the code, but it might work for Phase0 as well.
   Value innerring = iConfig.getParameter<int>("n_inner_ring_blades");
   Value outerring = max_value[intern("PXBlade")] - innerring;
-  auto& pxblade  = extractors[intern("PXBlade")];
+  auto pxblade  = extractors[intern("PXBlade")];
   addExtractor(intern("PXRing"), 
     [pxblade, innerring] (InterestingQuantities const& iq) {
       auto blade = pxblade(iq);
@@ -270,7 +270,7 @@ void GeometryInterface::loadFromTopology(edm::EventSetup const& iSetup, const ed
     }, 1, 2
   );
 
-  auto& pxmodule = extractors[intern("PXBModule")];
+  auto pxmodule = extractors[intern("PXBModule")];
   Value maxmodule = max_value[intern("PXBModule")];
   addExtractor(intern("HalfCylinder"),
     [pxendcap, pxblade, innerring, outerring] (InterestingQuantities const& iq) {

